tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -134,7 +153,7 @@ def test_user_defined_columns(self, dataframe: Any, local_remote_dir: Tuple[str,
             'out': out,
             'columns': user_defined_columns,
         }
-        with pytest.raises(ValueError, match=f'.* is not supported by MDSWriter.*'):
+        with pytest.raises(ValueError, match=f'.* is not supported by dataframe_to_mds.*'):
```

Comment:
could you add clarifying comment here as well? same as above

Reply:
Hi @XiaohanZhangCMU @snarayan21 this PR is very wanted :) , it looks the CI errors are explicits. 
Any chance to know when it will be integrated ? Thx

Reply:
@snarayan21 addressed the comments and added a "json" skipping, which I am not 100% is the best thing to do. 

@wolliq Sorry I have been busy with other stuff, if you are using the new types in some way, feel free to leave comments. I am not confident with the pr because I haven't figured out how to convert a general structtype(structfield,....) schema to MDS's json. The current workaround I put in is to skip schema checking when "json" is specified. Definitely not the perfect solution.